include/seqan3/alphabet/all.hpp

```diff
@@ -39,9 +39,121 @@
  *
  * \defgroup alphabet Alphabet
  *
- * The alphabet module contains different biological alphabets and related functionality.
+ * ## Introduction
+ *
+ * Alphabets are a core component in SeqAn, they enable us to represent the smallest
+ * unit of biological sequence data, e.g. a nucleotide or an amino acid.
+ *
+ * In theory, these could just be represented as a `char` and this is how many people
+ * perceive them, but it makes sense to use a smaller, stricter and well-defined alphabet
+ * in almost all cases, because:
```

Comment:
Maybe you could add a section about efficiency. Because, if I don't care about efficiency I could as well just use `char`s  and have some magic voodoo to ensure correctness. But our alphabets are not `char`s but different structs.
Convertible but not compatible.
And also one is merely interested in the char representation only in I/O and some debug information. Otherwise the program is just interested in the correct representation as a rank.

Reply:
I didn't want to add an extra section, but I have shifted focus a little and added more explanations.